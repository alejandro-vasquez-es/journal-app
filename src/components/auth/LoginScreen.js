import React from 'react';
import { useDispatch } from 'react-redux';

import { Link } from 'react-router-dom'
import { login, startLoginEmailPassword, startGoogleLogin } from '../../actions/auth';
import { useForm } from '../../hooks/useForms';

export const LoginScreen = () => {

	const dispatch = useDispatch();

	const [formValues, handleInputChange] = useForm({
		email: 'correo@gmail.com',
		password: 'test'
	});

	const { email, password } = formValues;

	const handleLogin = (e) => {
		e.preventDefault();
		dispatch(startLoginEmailPassword(email, password))
	}

	const handleGoogleLogin = () => {
		dispatch(startGoogleLogin());
	}

	return (
		<>
			<h2 className="auth__title">Login</h2>

			<form
				className="auth__form"
				onSubmit={handleLogin}
			>

				<input
					type="text"
					placeholder="email"
					name="email"
					className="auth__input"
					autoComplete='off'
					value={email}
					onChange={handleInputChange}
				/>

				<input
					type="password"
					placeholder="Password"
					name="password"
					className="auth__input"
					value={password}
					onChange={handleInputChange}
				/>

				<button
					className="button"
					type="submit"
				>
					Login
				</button>

				<div className="google__container">

					<p><b>Login with social networks </b></p>

					<div
						className="google-btn"
						onClick={handleGoogleLogin}
					>
						<div className="google-icon-wrapper">
							<img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
						</div>
						<p className="btn-text">
							<b>Sign in with google</b>
						</p>
					</div>

				</div>

				<Link
					className='link'
					to="/auth/register"
				>
					Create new account
				</Link>

			</form>

		</>
	)
}
