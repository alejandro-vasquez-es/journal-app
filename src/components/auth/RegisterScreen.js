import React from 'react'
import { Link } from 'react-router-dom'
import { useForm } from '../../hooks/useForms'

export const RegisterScreen = () => {

	// name: 'Alejandro',
	// email: 'correo@correo.com',
	// password: '123456',
	// password2: '123456' 

	const [formValues, handleInputChange] = useForm({
		name: 'Alejandro',
		email: 'correo@correo.com',
		password: '123456',
		password2: '123456'

	});

	const { name, email, password, password2 } = formValues;

	const handleLogin = (e) => {
		e.preventDefault();
		console.log(formValues);
	}

	const isFormValid = () => {
		//TODO:
	}

	return (
		<>
			<h2 className="auth__title">Register</h2>

			<form className="auth__form">

				<input
					type="text"
					placeholder="Name"
					name="name"
					className="auth__input"
					autoComplete='off'
					value={name}
					onChange={handleInputChange}
				/>

				<input
					type="text"
					placeholder="email"
					name="email"
					className="auth__input"
					autoComplete='off'
					value={email}
					onChange={handleInputChange}
				/>

				<input
					type="password"
					placeholder="Password"
					name="password"
					className="auth__input"
					value={password}
					onChange={handleInputChange}
				/>

				<input
					type="password"
					placeholder="Confirm password"
					name="password2"
					className="auth__input"
					value={password2}
					onChange={handleInputChange}
				/>

				<button
					className="button mb-5"
					type="submit"
					onClick={handleLogin}
				>
					Register
				</button>

				<Link
					className='link mt-1'
					to="/auth/login"
				>
					Already registered?
				</Link>

			</form>

		</>
	)
}
