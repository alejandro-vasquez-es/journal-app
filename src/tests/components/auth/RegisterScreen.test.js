import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { RegisterScreen } from '../../../components/auth/RegisterScreen';
import { mount } from 'enzyme';

import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { types } from '../../../types/types';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
	auth: {},
	ui: {
		loading: false,
		msgError: null
	}
};
let store = mockStore(initState);
const wrapper = mount(
	<Provider store={store}>
		<MemoryRouter>
			<RegisterScreen />
		</MemoryRouter>
	</Provider>
);

describe('Tests in <RegisterScreen />', () => {


	test('should display propertly', () => {

		expect(wrapper).toMatchSnapshot();

	});

	test('should dispatch the needed action', () => {

		wrapper.find('.auth__form').simulate('submit', {
			preventDefault() { }
		});

		const actions = store.getActions();

		expect(actions[0]).toEqual({
			type: types.uiSetError,
			payload: 'Name is require'
		})
	})

	test('should display the msgError div', () => {

		const errorMessage = 'Name is require';

		const initState = {
			auth: {},
			ui: {
				loading: false,
				msgError: errorMessage,
			}
		};
		const store = mockStore(initState);
		const wrapper = mount(
			<Provider store={store}>
				<MemoryRouter>
					<RegisterScreen />
				</MemoryRouter>
			</Provider>
		);

		expect(wrapper.find('.auth__alert-error').exists()).toBeTruthy();
		expect(wrapper.find('.auth__alert-error').text().trim()).toBe(errorMessage);

	})

})