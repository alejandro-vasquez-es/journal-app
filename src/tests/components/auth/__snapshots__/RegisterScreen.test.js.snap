// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <RegisterScreen /> should display propertly 1`] = `
Array [
  <h2
    className="auth__title"
  >
    Register
  </h2>,
  <form
    className="auth__form animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="text"
      value=""
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value=""
    />
    <button
      className="button mb-5"
      type="submit"
    >
      Register
    </button>
    <a
      className="link mt-1"
      href="/auth/login"
      onClick={[Function]}
    >
      Already registered?
    </a>
  </form>,
]
`;
